server/src/main/java/io/druid/server/QueryResource.java

```diff
@@ -183,6 +183,7 @@ public void write(OutputStream outputStream) throws IOException, WebApplicationE
                   {
                     // json serializer will always close the yielder
                     jsonWriter.writeValue(outputStream, yielder);
+                    outputStream.flush(); // Some types of OutputStream suppress flush errors in the .close() method.
```

Comment:
just curious, if there is no evidence then why are we fixing it now?
Usual usage pattern of OutputStream I have seen is to just rely on close call unless the usecase really demanded intermediate flushes while you write data and then write more after the flush and so on.